das/worker.go

```diff
@@ -0,0 +1,103 @@
+package das
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/hashicorp/go-multierror"
+)
+
+type job struct {
+	id           int
+	from, to     uint64
+	fromPriority bool
+}
+
+func (s *state) newJob(from, max uint64, fromPriority bool) job {
+	s.nextJobID++
+	to := from + s.rangeSize - 1
+	if to > max {
+		to = max
+	}
+	return job{
```

Comment:
Same here. Lot's of jobs are allocated during runtime, not sure why the needs to be copied while being passed through each method boundary

Reply:
Answered in upper comment